### ISSUE TITLE
Insert Link: up/down arrow in link autocomplete crashes block

### ISSUE THREAD

**Issue by annestk** [WordPress/gutenberg#10689]
**Describe the bug**
Pressing the up or down arrow in the link autocomplete field crashes the block

**To Reproduce**
1. In a paragraph, select text
2. Click add link
3. Search to trigger the autocomplete dropdown
4. Press the up or down arrow key (regardless of whether the autocomplete suggestions is above or below)

**Expected behavior**
Up or down arrow key should move focus to an item on the list of suggestions.

I saw issue #10496 with the fix #10500 should be included in 4.0 (was it?), but clicking an autocomplete suggestion doesn't insert a link, it just selects a block or two.

With neither mouse or keyboard selection working, it's not possible to insert link using the autocompleter.

**Screenshots**
![screenshot 22](https://user-images.githubusercontent.com/38467445/47097712-45708580-d232-11e8-92a1-b163cdc06bfc.png)
![screenshot 23](https://user-images.githubusercontent.com/38467445/47097725-49040c80-d232-11e8-87c9-86445524515e.png)

**Console error**
```
TypeError: Cannot read property 'body' of undefined
    at index.js?ver=1539782380:12
    at s (index.js?ver=1539782380:12)
    at Object.offset (index.js?ver=1539782380:12)
    at e.exports (index.js?ver=1539782380:12)
    at t.value (index.js?ver=1539782380:50)
    at jc (react-dom.min.82e21c65.js:134)
    at gc (react-dom.min.82e21c65.js:127)
    at vb (react-dom.min.82e21c65.js:126)
    at ub (react-dom.min.82e21c65.js:126)
    at wg (react-dom.min.82e21c65.js:137)
```

**Desktop:**
 - OS: Windows 10
 - Browser Chrome
 - Version WP 4.9.8, GB 4.0

**Comment by designsimply** [comment]
Good find! Tested and confirmed using WordPress 4.9.8 and Gutenberg 4.0 (which was just released) with Firefox 62.0.3 on macOS 10.13.6.

![screen shot 2018-10-17 at 11 49 18 am](https://user-images.githubusercontent.com/1119271/47106032-cd3e9b80-d202-11e8-8bc2-a2eb660464b8.png)

**Comment by talldan** [comment]
Thanks for the report @annestk - this does seem to be related to my fix for #10500 😞. Working on a fix for the fix now.

**Comment by talldan** [comment]
The fix has now been merged @annestk - thanks again for catching this so soon after the release.

I'm not sure whether there will be an intermediate 4.0.1 release of the plugin, but the fix will make its way into 4.1 at the very least.
